runtime/compiler/ilgen/Walker.cpp

```diff
@@ -7266,26 +7247,18 @@ TR_J9ByteCodeIlGenerator::storeStatic(int32_t cpIndex)
    _staticFieldReferenceEncountered = true;
    TR::Node * value = pop();
 
-   if (TR::Compiler->om.areFlattenableValueTypesEnabled())
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+   if (owningMethod->isFieldNullRestricted(comp(), cpIndex, true /* isStatic */, true /* isStore */))
       {
-      TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
-
-      if (!TR::Compiler->om.isQDescriptorForValueTypesSupported())
+      if (comp()->getOption(TR_TraceILGen))
          {
-         TR_ASSERT_FATAL(false, "Support for null-restricted types without Q descriptor is to be implemented!!!");
+         traceMsg(comp(), "%s: cpIndex %d isFieldNullRestricted 1 value %d isNonNull %d\n", __FUNCTION__, cpIndex, value->getGlobalIndex(), value->isNonNull());
```

Comment:
I know the message string was that way originally, but may I ask you to change `value %d` to `value n%dn`, is it's printing a node's index?

Reply:
Addressed in 32f6437